#!/usr/bin/env python

PACKAGE = "range_image_segment"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("voxel_grid_size",        double_t, 0, "Default: 0.1",    0.1,  0.0,  0.5)

gen.add("detect_min",             double_t, 0, "Default: 1",      1,    0,  2)
gen.add("detect_max",             double_t, 0, "Default: 4",      4,    2,  12)

gen.add("CorePointMinPt",         int_t, 0,    "Default: 10",      10,   2,   100)
gen.add("ClusterTolerance",       double_t, 0, "Default: 0.3",     0.3,   0,   1)
gen.add("MinClusterSize",         int_t, 0,    "Default: 30",      30,   10,   100)
gen.add("MaxClusterSize",         int_t, 0,    "Default: 1000",    1000,   20,   10000)

gen.add("segmentTh_H",           double_t, 0, "Default: 20",     20,    0,  90)
gen.add("segmentTh_V",           double_t, 0, "Default: 20",     20,    0,  90)

gen.add("d_th",                   double_t, 0, "Default: 0.1",    0.1,    0,  1)


gen.add("roi_max_x",              double_t, 0, "Default: 6",     6,   4,    50)
gen.add("roi_max_y",              double_t, 0, "Default: 3",     3,   2,    30)
gen.add("roi_max_z",              double_t, 0, "Default: 1",     1,   0,    3)

gen.add("roi_min_x",              double_t, 0, "Default: -4",    -4,  -30, 0)
gen.add("roi_min_y",              double_t, 0, "Default: -3",   -3,  -30, -2)
gen.add("roi_min_z",              double_t, 0, "Default: -3",    -3,   -5,   0)

exit(gen.generate(PACKAGE, "range_image_segment", "range_image_segment_"))